File: fanda/src/services/remote/recordService.js
import { get, post, del } from '@/utils/http'
export default {
  async getAll() { return get('/api/records') },
  async getByDate(date) { return get('/api/records', { date }) },
  async getByMonth(year, month) { return get('/api/records', { year, month }) },
  async save(record) { return post('/api/records', record) },
  async delete(id) { return del('/api/records/' + id) },
}

File: fanda-server/src/main/java/com/fanda/service/RecordService.java
package com.fanda.service;

import com.fanda.dto.request.RecordCreateRequest;
import com.fanda.entity.MealRecord;

import java.time.LocalDate;
import java.util.List;

public interface RecordService {
    List<MealRecord> getAll(Long userId);
    List<MealRecord> getByDate(Long userId, LocalDate date);
    List<MealRecord> getByMonth(Long userId, int year, int month);
    MealRecord create(RecordCreateRequest request, Long userId);
    void delete(Long id, Long userId);
}
